native/src/windows.rs

```diff
@@ -1,46 +1,69 @@
 use neon::prelude::*;
-use winapi::shared::minwindef::{TRUE, FALSE};
-
-use winapi::um::consoleapi::*;
-use winapi::um::wincon::*;
-use std::process::Command;
+use std::process::{Command};
+use winapi::um::{wincon, consoleapi};
 
 // https://gist.github.com/rdp/f51fb274d69c5c31b6be
 
+fn free_console() -> Result<(), String> {
+    let result = unsafe { wincon::FreeConsole() };
+    match result {
+        0 => Ok(()),
+        _ => Err(format!("unable to free console"))
+    }
+}
+
+fn attach_console(pid: u32) -> Result<(), String> {
+    let result = unsafe { wincon::AttachConsole(pid) };
+    match result {
+        0 => Ok(()),
+        _ => Err(format!("unable to attach to console of {pid}", pid=pid))
+    }
+}
+
+fn clear_console_ctrl_handler(add: bool) -> Result<(), String> {
+    let result = unsafe { consoleapi::SetConsoleCtrlHandler(None, add.into()) };
+    match result {
+        0 => Ok(()),
+        _ => Err(format!("unable to clear ctrls handler"))
+    }
+}
+
+fn generate_ctrlc_event() -> Result<(), String> {
+    let result = unsafe { wincon::GenerateConsoleCtrlEvent(wincon::CTRL_C_EVENT, 0) };
+    match result {
+        0 => Ok(()),
+        _ => Err(format!("unable to generate ctrls event"))
+    }
+}
+fn ctrlc_perform(pid: u32, holder_pid: u32) -> Result<(), String> {
+    free_console()?; // free ourselves from our own console
+    attach_console(pid)?; // attach to the console of the process we want to send ctrl-c to
+    clear_console_ctrl_handler(true)?; // remove our own ctrl-c handler so we don't kill ourselves
+    generate_ctrlc_event()?; // generate the ctrlc event in our own console, which is now the console of the target process
+    free_console()?; // free ourselves from out own console again
+    attach_console(holder_pid)?; // attach to the console of the holder process
+    clear_console_ctrl_handler(false)?; // what does this do?!
```

Comment:
```suggestion
    clear_console_ctrl_handler(false)?; // reinstall the default ctrl-c handler which is to close this process
```

Reply:
I tried to figure out if there was a way to get the _current_ console cttrl handler so that we could reinstall _that_ whatever it was, but no dice.